src/test/java/de/qabel/desktop/hockeyapp/VersionClientTest.java

```diff
@@ -0,0 +1,200 @@
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONException;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class VersionClientTest {
+
+    private static final int VERSION_ID_1_1 = 208;
+    private static final int VERSION_ID_1_0 = 195;
+
+    CloseableHttpClientStub httpClient = new CloseableHttpClientStub();
+    private HockeyAppConfiguration config = new HockeyAppConfiguration("1.1", httpClient);
+    private VersionClient client = new VersionClient(config, httpClient);
+
+    @Test
+    public void checkAppVersion() {
+        assertEquals("1.1", config.getAppVersion());
+    }
+
+    @Test
+    public void findVersion() throws VersionNotFoundException, IOException {
+
+        String shortVersion = "1.1";
+        buildTestVersions();
+        HockeyAppVersion version = client.getVersion();
+
+        assertEquals(shortVersion, version.getShortVersion());
+    }
+
+    @Test(expected = IOException.class)
+    public void parseInvalidVersionCreateResponse() throws IOException {
+        String responseContent = "nbzuhbggzubzug";
+        client.parseVersionCreateResponse(responseContent);
+        client.createVersion("invalid");
+    }
+
+    @Test(expected = IOException.class)
+    public void parseInvalidVersionsResponse() throws IOException {
+        String responseContent = "nbzuhbggzubzug";
+        client.parseVersionsResponse(responseContent);
+    }
+
+    @Test
+    public void parseVersionCreateResponse() throws IOException {
+
+        String versionsResponseContent = getVersionCreateResponseString(config.getAppVersion());
+
+        client.parseVersionCreateResponse(versionsResponseContent);
+
+        assertEquals(VERSION_ID_1_1, client.getVersion().getVersionId());
+    }
+
+    @Test
+    public void parseVersionsResponse() throws IOException {
+        String versionsResponseContent = getVersionsJsonString();
+        List<HockeyAppVersion> versions = client.parseVersionsResponse(versionsResponseContent);
+        assertEquals(2, versions.size());
+    }
+
+    @Test
+    public void createNewVersion() throws IOException {
+
+        String shortVersion = "1.5";
+        buildTestVersions();
+
+        loadFakeCreatedVersionResponse(shortVersion);
+        client.findAndLoadVersion(shortVersion);
+
+        assertEquals(shortVersion, client.getVersion().getShortVersion());
+
+    }
+
+    @Test
+    public void loadVersions() throws IOException, JSONException {
+        loadFakeVersions();
+        List<HockeyAppVersion> versions = client.getVersions();
+
+        assertEquals(2, versions.size());
+
+        assertEquals(VERSION_ID_1_1, versions.get(0).getVersionId());
+        assertEquals(VERSION_ID_1_0, versions.get(1).getVersionId());
+
+        assertEquals("1.1", versions.get(0).getShortVersion());
+        assertEquals("1.0", versions.get(1).getShortVersion());
```

Comment:
`assertShortVersion("1.0", versions.get(1))`
You should refactor your test too and for example extract such an assertion here to reduce code duplication.